### PR TITLE
2.0.0 -> master

### PR DESCRIPTION
**Pull request by ovr**
Hey!

Type: bug fix | new feature | code quality | documentation

Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/SocialConnect/auth/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

Thanks :smiley_cat:

### CODE REVIEW

**Comment by ovr** [comment]
@heiglandreas @ADmad @s0lus If you have an ideas what we can break in 2.0.0, it's the best time to say me about it ;)

File: example/config.php.dist [hunk 1]
```diff
@@ -40,7 +50,14 @@ return [
         'vk' => [
             'applicationId' => '',
             'applicationSecret' => '',
-            'scope' => ['email']
+            'scope' => ['email'],
+            'options' => [
+                'identity.fields' => array(
```
[the remark this answers is not in the document]
Comment (ovr) [reply example/config.php.dist]: Nice catch 👍

File: src/OpenIDConnect/AbstractProvider.php [hunk 2]
```diff
@@ -86,19 +87,20 @@ public function getJWKSet()
...
-            //'response_mode' => 'form_post',
-            'scope' => 'openid'
-        ];
+        $parameters = parent::getAuthUrlParameters();
+
+        // special parameters only required for OpenIDConnect
+        $parameters['client_id'] = $this->consumer->getKey();
+        $parameters['redirect_uri'] = $this->getRedirectUrl();
+        $parameters['response_type'] = 'code';
+        $parameters['response_mode'] = 'form_post';
```
[the remark this answers is not in the document]
Comment (ovr) [reply src/OpenIDConnect/AbstractProvider.php]: Oops! 😸 Nice catch 👍

**Comment by ADmad** [comment]
@ovr A feature that would be nice to have in new version of this lib is, being able to get access to the http client used by provider with the accesss token setup. This would allow to easily make additional API requests to the provider.

For e.g. let's say my site allow auth using Facebook and I also have to fetch the friends list of authenticated user. Having access to the http client of Facebook provider would allow me to easily make an API query to fetch friends list which additional effort of crafting a separate http client.

**Pull request merged**